lib/inlines.js

```diff
@@ -130,9 +130,6 @@ var spnl = function() {
 // literal sequence of backticks.
 var parseBackticks = function(block) {
     var ticks = this.match(reTicksHere);
-    if (ticks === null) {
-        return false;
-    }
     var afterOpenTicks = this.pos;
```

Comment:
Can you explain why this is redundant?

Reply:
Function `parseBackticks()` be called from `parseInline()` only, which has peeked an exact byte `` ` `` from `this.subject`. Function `match()` also test regex from `this.subject`, so I think this is a redundant check. 

Function `peek()` get a byte from the current pos: 
```js
var peek = function() {
    if (this.pos < this.subject.length) {
        return this.subject.charCodeAt(this.pos);
    } else {
        return -1;
    }
};
```

`parseBackticks()`'s caller `parseInline()` hits case `C_BACKTICK`:
```js
    case C_BACKTICK:
        res = this.parseBackticks(block);
        break;
```

`match()` test it at the current pos, it will hit regex  `reTicksHere`:

```js
var match = function(re) {
    var m = re.exec(this.subject.slice(this.pos));
    if (m === null) {
        return null;
    } else {
        this.pos += m.index + m[0].length;
        return m[0];
    }
};
```

So I think this is a redundant check, but I am not very sure if there are other situations. If I am wrong, please help me figure it out, thank you.

Reply:
I think you're probably right. On the other hand, this seems a good safety check.  Is there a measurable performance impact to removing this?  I would be surprised if there were, so I'm inclined to leave it in as a safety measure. After all, the calling context may change at some point in the future.

Reply:
Remove the null check can not help performance better in millisecond-scale. Thank you for your reply, I can understand it, so I have reverted it. 

Please review other changes, thank you.